docs/developer/payments/price-calculations.mdx

```diff
@@ -0,0 +1,138 @@
+  - Grouping ensures that events of the same type and PSP reference are properly matched to calculate amounts accurately.
+#### 2. Pending Amounts:
+  - pending amounts (`authorizePendingAmount`, `chargePendingAmount`, `refundPendingAmount`, `cancelPendingAmount`) are increased only if a `REQUEST` event exists for the corresponding PSP reference.
+  - If a success or failure event is also associated with the same PSP reference, the system assumes the requested amount has already been processed, and the pending amount will not be increased.
+#### 3. `SUCCESS` and `FAILURE` Events:
+  - The transaction amount (e.g., `authorizedAmount`, `chargedAmount`) is increased only when a `SUCCESS` event exists.
+  - If both a `SUCCESS` and `FAILURE` event are present for the same PSP reference, the system compares their creation timestamps:
+	- If the `FAILURE` event is newer, the `SUCCESS` event is ignored.
+	- If the `SUCCESS` event is newer, it is used for the calculations.
+#### 4. Events Without PSP References:
+  - Some events (e.g., those created by the system to maintain correct amounts) may not have a PSP reference.
+  - The events of the following types are still included in the transaction amounts and influence the calculated values:
+    - `AUTHORIZATION_SUCCESS`
+    - `AUTHORIZATION_ADJUSTMENT`
+    - `CHARGE_SUCCESS`
+    - `CHARGE_BACK`
+    - `REFUND_SUCCESS`
+    - `REFUND_REVERSE`
+    - `CANCEL_SUCCESS`
+#### 5. Adjustment Events:
+  - The `AUTHORIZATION_ADJUSTMENT`, overwrite previous amounts, if an adjustment is present,
+  older authorization events are ignored, and the adjusted amount becomes the new authorized value.
+
+This logic ensures that transaction amounts reflect the current state of the transaction based on its event history while accounting for both processed and pending actions.
+
+### Calculation Details
+
+When the transaction amounts are recalculated, firstly all values are set to zero, 
+then the calculations are applied in the following order:
+  1. recalculate amounts based on events without PSP reference
+  2. recalculate amounts based on `AUTHORIZATION` events
+  3. recalculate amounts based on `CHARGE` events
+  4. recalculate amounts based on `REFUND` events
+  5. recalculate amounts based on `CANCEL` events
+
+#### Authorized Value ([`transactionItem.authorizedAmount`](/api-reference/payments/objects/transaction-item#transactionitemauthorizedamountmoney---))
+  - When an `AUTHORIZATION_SUCCESS` event occurs and there is no `AUTHORIZATION_FAILURE` or the `FAILURE` event is older: value increases by the `SUCCESS` event's amount.
+  - When an `AUTHORIZATION_ADJUSTMENT` event occurs: value is overwritten by the adjustment's amount.
+  - When a `CHARGE_REQUEST` or `CHARGE_SUCCESS` event occurs: value is reduced by the event's amount. 
+  - When a `CANCEL_REQUEST` or `CANCEL_SUCCESS` event occurs: value is reduced by the event's amount. 
+  - The amount cannot be negative, the minimum value is 0.
```

Comment:
suggestion: I think we should use given > when > then format for such description, and avoid using "occurs" for events, because we recalculate history on each change.

This way I thin that I've found we are missing some info in this paragraph:
- **WHEN** CHARGE_REQUEST and CHARGE_SUCCESS reduce `authorizedAmount`: I think such event must be **after** authorization and they both must have the same pspReference? I also think there cannot be `CHARGE_FAILURE` event?
- **WHEN** CANCEL_REQUEST and CANCEL_SUCCESs reduce `authorizedAmount`: same thing, it must be **after** authorization event? and there cannot be CANCEL_FAILURE event?

```suggestion
  - **GIVEN** there is an `AUTHORIZATION_SUCCESS` event
    - **WHEN there is NO** `AUTHORIZATION_FAILURE`
    - **OR WHEN** the `AUTHORIZATION_FAILURE` event is older than `AUTHORIZATION_SUCCESS`
      - **THEN** value increases by the `AUTHORIZATION_SUCCESS` event's amount.
  - **GIVEN** an `AUTHORIZATION_ADJUSTMENT` event
    - **WHEN** it's the latest event of this type
      - **THEN** value is overwritten by the `AUTHORIZATION_ADJUSTMENT` event amount.
  - **GIVEN** `CHARGE_REQUEST` or `CHARGE_SUCCESS` event:
    - **THEN** value is reduced by the event's amount. 
  - **GIVEN** `CANCEL_REQUEST` or `CANCEL_SUCCESS` event:
    - **THEN** value is reduced by the event's amount.
  - **GIVEN** all calculations were performed and we have result `authorizationAmount`
    - **WHEN** calculated `authorizationAmount` is below `0`
  	  - **THEN** `authorizationAmount` is set to `0` (it cannot be lower than 0)
```

Reply:
Done